internal/viridian/cluster.go

```diff
@@ -7,34 +7,49 @@ import (
 	"math/rand"
 	"os"
 	"path"
+	"strings"
 	"time"
 )
 
 type createClusterRequest struct {
-	KubernetesClusterID int         `json:"kubernetesClusterId"`
-	Name                string      `json:"name"`
-	ClusterTypeID       ClusterType `json:"clusterTypeId"`
-	PlanName            ClusterPlan `json:"planName"`
+	KubernetesClusterID int    `json:"kubernetesClusterId"`
+	Name                string `json:"name"`
+	ClusterTypeID       int64  `json:"clusterTypeId"`
+	PlanName            string `json:"planName"`
 }
 
 type createClusterResponse Cluster
 
-func (a API) CreateCluster(ctx context.Context, name string, plan ClusterPlan, k8sClusterID int, isDev bool) (Cluster, error) {
+const defaultClusterType = "devmode"
+
+func (a API) CreateCluster(ctx context.Context, name string, clusterType string, k8sClusterID int) (Cluster, error) {
 	if name == "" {
 		name = clusterName()
 	}
-	if plan == "" {
-		plan = ClusterPlanServerless
+	cType, err := a.FindClusterType(ctx, clusterType)
+	if err != nil {
+		return Cluster{}, err
 	}
-	ct := ClusterTypeProd
-	if isDev {
-		ct = ClusterTypeDev
+	clusterTypeID := cType.ID
+	planName := cType.Name
+	if cType.Name == "" {
+		clusterTypes, err := a.ListClusterTypes(ctx)
+		if err != nil {
+			return Cluster{}, err
+		}
+		for _, ct := range clusterTypes {
+			if strings.ToLower(ct.Name) == defaultClusterType {
+				clusterTypeID = ct.ID
+				planName = ct.Name
+				break
```

Comment:
What happens if the loop ends without setting clusterTypeID and planName?

Reply:
this if block become useless when we set DEVMODE as default. So, I am removing it.